### PR TITLE
tests: run_node should retry when it meets gRPC bind fail

### LINKED ISSUE #2905: test: test_multi_server_cluster_restart sometimes fails in CI
https://circleci.com/gh/pingcap/tikv/9107

### PR DESCRIPTION
**Pull request by overvenus**
Close #2905

### CODE REVIEW

**Comment by breeswish** [comment]
How about using an exponential retry delay? i.e. 1sec, 2sec, 4sec, 8sec

File: tests/raftstore/server.rs [hunk 1]
```diff
@@ -147,18 +149,27 @@ impl Simulator for ServerCluster {
...
+                snap_mgr.clone(),
+                pd_worker.scheduler(),
+                Some(engines.clone()),
+                Some(import_service.clone()),
+            ));
+            if let Some(Err(Error::Grpc(GrpcError::BindFail(ref addr, ref port)))) = server {
+                debug!("fail to create a server: bind fail {:?}", (addr, port));
+                thread::sleep(Duration::from_millis(100));
+                continue;
+            }
```
Comment (breeswish) [inline tests/raftstore/server.rs]: If it is not a `BindFail`, it should panic immediately.

Comment (hicqu) [reply tests/raftstore/server.rs]: And I think we can eliminate the `Option`.

**Comment by overvenus** [comment]
> How about using an exponential retry delay? i.e. 1sec, 2sec, 4sec, 8sec. Existing library can be used.

I think a simple backoff is enough.

> And I think we can eliminate the Option.

It's a `for` loop, so down below may use an initialized value.

File: tests/raftstore/server.rs [hunk 2]
```diff
@@ -147,18 +149,32 @@ impl Simulator for ServerCluster {
...
+                sim_router.clone(),
+                resolver.clone(),
+                snap_mgr.clone(),
+                pd_worker.scheduler(),
+                Some(engines.clone()),
+                Some(import_service.clone()),
+            ));
+            match server {
+                Some(Ok(_)) => break,
+                Some(Err(Error::Grpc(GrpcError::BindFail(ref addr, ref port)))) => {
```
Comment (huachaohuang) [inline tests/raftstore/server.rs]: Please add a comment here to explain when can this happen.

**Comment by breeswish** [comment]
LGTM

File: tests/raftstore/server.rs (hunk 2 above)
Comment (overvenus) [reply tests/raftstore/server.rs]: Done

**Pull request merged**